invenio_rdm_records/services/generators.py

```diff
@@ -247,3 +248,19 @@ def needs(self, record=None, **kwargs):
     def query_filter(self, identity=None, **kwargs):
         """Filters for current identity as member."""
         return dsl.Q("terms", **{"parent.communities.ids": self.communities(identity)})
+
+
+class IfConfig(ConditionalGenerator):
```

Comment:
should this be in invenio-records-resources or even invenio-records-permissions? It could be useful for features in other modules (e.g. communities)

Reply:
I agree, since it's a step towards lightening RDM-Records (we should always aim for this), but I would shelve for now since there's a bigger chain of changes that we probably don't want to get hit with now.